internal/controller/providers/pg/provider.go

```diff
@@ -147,6 +149,32 @@ func (p *Provider) Status(ctx context.Context) (everestv1alpha1.DatabaseClusterS
 		status.Status = everestv1alpha1.AppStateRestoring
 	}
 
+	// If the PVC resize is currently in progress, or just finished, we need to
+	// check if it failed in order to set or clear the error condition.
+	if status.Status == everestv1alpha1.AppStateResizingVolumes ||
```

Comment:
why do we check DBCluster.Status instead of p.PerconaPGCluster ? Can we determine that p.PerconaPGCluster is in state "Storage resizing"?

Reply:
We check the status because we should have calculated it just before this block but I just found out that I have them in the incorrect order.
I just pushed https://github.com/percona/everest-operator/pull/718/commits/ab73bf579b30376d2550be5dc6d87726bfeb945c to fix this.

Reply:
Because the underlying CR doesn't have the info we need, we need to look into STSs, PVCs and other objects. Please have a look at how that status is calculated for each provider:
* [PXC](https://github.com/percona/everest-operator/blob/7c6a6a61ada391834bed66b88aaf35ec28cce4b4/internal/controller/providers/pxc/provider.go#L230-L234)
* [PSMDB](https://github.com/percona/everest-operator/blob/7c6a6a61ada391834bed66b88aaf35ec28cce4b4/internal/controller/providers/psmdb/provider.go#L163-L188)
* [PG](https://github.com/percona/everest-operator/blob/7c6a6a61ada391834bed66b88aaf35ec28cce4b4/internal/controller/providers/pg/provider.go#L171-L204)